src/main/kotlin/com/redhat/devtools/intellij/kubernetes/editor/NonClusterResourceEditor.kt

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.kubernetes.editor
+
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.project.Project
+import com.redhat.devtools.intellij.kubernetes.model.IResourceModel
+import io.fabric8.kubernetes.api.model.HasMetadata
+
+class NonClusterResourceEditor(
```

Comment:
On a conceptual level I am wondering if it is a good choice to extend the existing `ResourceEditor` for your editor. It might be a better idea to create a separate editor and aggregate features that you may reuse. Especially if you dont need the `ClusterResource`, many features of the editor are of no use to you.
What are the features of the `ResourceEditor` that you need?
Thoughts?

ps. I'll skip the changes in `ResourceEditor` and it's factory until I can come up with informed approval or suggestions on how to allow you to reuse features in your editor(s).

ps2. #549 changes a lot in the `ResourceEditor` by providing support for multiple resources. We'll need additional work for sync up #549 with this PR.